### ISSUE TITLE
Some questions

### ISSUE THREAD

**Issue by prsolucoes** [Tencent/rapidjson#1163]
Hi,

I have some question. Can you help me?

1 - To get a default what is the process?

```
const rapidjson::Value& username = args.HasMember("username") ? args["username"] : (?what?) ;
```

2 - How to fast/correct check for a null or non-exists tag?

```
if (args.HasMember("username") && args.HasMember("password")) {
    const rapidjson::Value& username = args["username"];
    const rapidjson::Value& password = args["password"];
}

// or i can do:
const rapidjson::Value& username = args["username"].IsNull() ? (??what??) : args["username"];            
```

What i can put on "(??what??)" ?

Thanks.

**Comment by hammersmith-xie** [comment]
try create a null value.
`rapidjson::Value  tmpValue;
    const rapidjson::Value &username = (document.HasMember("username") && document["username"].IsString())?document["username"]:tmpValue;
    if(pvalue.IsNull())
    {
        return 0;
    }`

**Comment by prsolucoes** [comment]
Yes, is this. But how i can pass "rapidjson::Value tmpValue" to the ternary line? Something like this:

```
const rapidjson::Value &username = (document.HasMember("username") && document["username"].IsString())?document["username"]: rapidjson::Value();
```

But my example dont work.

Look my class:

https://github.com/ezored/dependency-sample/blob/master/build/source/cpp/CustomerSystemService.cpp#L62

If nested a lot.

**Comment by pah** [comment]
For such a nested lookup, you might want to check out the [JSON Pointer](http://rapidjson.org/md_doc_pointer.html) API.
